src/Tooltip.js

```diff
@@ -50,23 +58,25 @@ class Tooltip extends React.Component {
   }
 
   onMouseOverTooltip() {
-    if (this._hoverTimeout) {
-      clearTimeout(this._hoverTimeout);
-    }
-
-    if (!this.props.isOpen) {
-      this.toggle();
+    if (this._hideTimeout) {
+      clearTimeout(this._hideTimeout);
     }
+    this._showTimeout = setTimeout(this.show, this.getDelay('show'));
   }
 
   onMouseLeaveTooltip() {
-    this._hoverTimeout = setTimeout(this.onTimeout, 250);
+    if (this._showTimeout) {
+      clearTimeout(this._showTimeout);
+    }
+    this._hideTimeout = setTimeout(this.hide, this.getDelay('hide'));
   }
 
-  onTimeout() {
-    if (this.props.isOpen) {
-      this.toggle();
+  getDelay(key) {
```

Comment:
It's true that that would reduce some overhead, but I feel the tiny performance increase would not be worth complicating the code with extra lifecycle logic.

It's only a couple of direct accesses from the props and an if statement, and `mouseover` and `mouseleave` events aren't realistically going to happen very often unless the user is flailing the mouse around.